src/adapter/src/catalog.rs

```diff
@@ -3500,6 +3500,15 @@ impl<S: Append> Catalog<S> {
 
                     let entry = state.get_entry(&id);
                     let name = entry.name().clone();
+
+                    if let ResolvedDatabaseSpecifier::Ambient = name.qualifiers.database_spec {
```

Comment:
Good call, that seems much better, do you think I should also change the existing `Op::CreateItem` error logic?

Reply:
Oh I didn't realize that's how CreateItem works. I'm not sure if one of those is better, and/or if they match exactly the same set of items. Maybe ambient databases are all system ids? Not sure. Are there non-system objects that have ambient dbs? Are there user things with ambient dbs?

Reply:
Oh wait, CreateItem doesn't have an ID because it hasn't been created yet. So I think we just leave that one alone and change the others to your suggestion.

Temporary items also have ambient IDs which is another reason that your suggestion is better.